pkg/steps/multi_stage.go

```diff
@@ -407,6 +408,15 @@ func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []cor
 		artifactDir := step.ArtifactDir
 		if s.artifactsViaPodUtils {
 			artifactDir = fmt.Sprintf("%s/%s", s.name, step.As)
+			gracePeriod := 30 * time.Second
+			if step.TerminationGracePeriodSeconds != nil {
+				gracePeriod = time.Duration(*step.TerminationGracePeriodSeconds) * time.Second
+			}
+			// We want upload to have some time to do what it needs to do, so set
+			// the grace period for the process to be just shy of that, assuming
+			// an 80/20 distribution of work.
+			gracePeriod = (gracePeriod * 4) / 5
```

Comment:
Isn't this redundant with the test-infra change?

Reply:
No wait, we are removing it here again? Why?

Reply:
No, for two reasons:

 1. `test-infra` controls the `ci-operator` Pod that we start for the Prow job, and the user controls there the execution time and grace period of the test process (we increase the amount to set the grace period of the Pod)
 2. `ci-tools` here controls the test pods for a step in a workflow, and the user today sets the overall grace period for the Pod (will fix this), so we reduce the given time to get the grace period for the process

Reply:
(So the two parts of the code are unrelated and act on separate Pods with separate user inputs. My goal is to make users set timeout/grace period on step pods the same way in the future)